library/RemoteServer.cpp

```diff
@@ -476,11 +476,16 @@ void ServerMainImpl::threadFn(std::promise<bool> promise, int port)
     CActiveSocket *client = nullptr;
 
     try {
-        while ((client = server.socket.Accept()) != NULL)
+        while (server.socket.IsSocketValid())
         {
-            BlockGuard lock;
-            ServerConnection::Accepted(client);
-            client = nullptr;
+            if ((client = server.socket.Accept()) != NULL)
+            {
+                BlockGuard lock;
+                ServerConnection::Accepted(client);
+                client = nullptr;
+            } else {
+                std::cerr << "Accepting connection error: " << server.socket.DescribeError() << std::endl;
+            }
         }
     } catch(BlockedException &) {
```

Comment:
This should be done with the debug logging system so the output verbosity is filterable.

```cpp
WARN(socket).print("Accepting connection error: %s\n", server.socket.DescribeError());
```

Might need a `.c_str()` in there if the error comes back as a std::string.

You'll need to include Debug.h and set up the category at the top of the file:

```cpp
namespace DFHack {
    DBG_DECLARE(core, socket, DebugCategory::LINFO);
}
```

Reply:
Updated the merge request, left all the other outputs in there. Imagine they would be better in another merge request.